src/App.tsx

```diff
@@ -1,29 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+enum PerPageAmount {
+  Three = 3,
+  Five = 5,
+  Ten = 10,
+  Twenty = 20,
+}
+
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(PerPageAmount.Five);
+
+  const totalAmount = items.length;
+  const firstIndex = (currentPage - 1) * perPage;
+  const lastIndex = Math.min(currentPage * perPage, totalAmount);
+  const currentItems = items.slice(firstIndex, lastIndex);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {firstIndex + 1} - {lastIndex} of{' '}
+        {totalAmount})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={perPage}
+            onChange={event => {
+              setCurrentPage(1);
+              setPerPage(+event.target.value);
+            }}
+          >
+            <option value="3">{PerPageAmount.Three}</option>
+            <option value="5">{PerPageAmount.Five}</option>
+            <option value="10">{PerPageAmount.Ten}</option>
+            <option value="20">{PerPageAmount.Twenty}</option>
```

Comment:
```suggestion
            {[3, 5, 10, 20].map(option => <option key={option} value={option}>{option}</option>)}
```
looks like redundant to make enum for options.
suggest to make array and map all options.
something like that